tests/testthat/test_model_conversions.R

```diff
@@ -0,0 +1,71 @@
+context("Unit Conversion Refactoring Tests")
+
+# These tests validate the refactoring to use PEcAn.utils::ud_convert()
+# for consistent unit conversions across SIPNET, DALEC, GDAY, and FATES models.
+
+test_that("UDUNITS2 recognizes new unit strings", {
+  library(PEcAn.utils)
+  
+  # Pool conversions (mass only)
+  expect_equal(ud_convert(100, "g/m2", "kg/m2"), 0.1)
+  expect_equal(ud_convert(1000, "g/m2", "kg/m2"), 1.0)
+  
+  # Flux conversions (mass and time)
+  expect_equal(ud_convert(86400, "g/m2/d", "kg/m2/s"), 0.001, tolerance = 1e-10)
+  
+  # Large-scale conversions (GDAY)
+  expect_equal(ud_convert(10, "Mg/ha", "kg/m2"), 1.0)
+  
+  # Energy conversions (FATES)
+  expect_equal(ud_convert(1000, "J/mol", "kJ/mol"), 1.0)
+})
+
+test_that("SIPNET flux conversions produce realistic values", {
+  library(PEcAn.utils)
+  
+  # GPP: ~5 g C/m2/d → kg/m2/s (should be very small)
+  gpp_result <- ud_convert(5, "g/m2/d", "kg/m2/s")
+  expect_true(gpp_result > 0)
+  expect_true(gpp_result < 5)  # Should be much smaller
```

Comment:
This is far too wide a range -- we need this result to be correct within float error, which is what expect_equal enforces.
```suggestion
  # GPP: ~5 g C/m2/d --> kg/m2/s
  expect_equal(ud_convert(5, "g/m2/d", "kg/m2/s"), 5.79e-8)
```